airflow/providers/http/hooks/http.py

```diff
@@ -340,10 +340,7 @@ async def run(
         if headers:
             _headers.update(headers)
 
-        if self.base_url and not self.base_url.endswith("/") and endpoint and not endpoint.startswith("/"):
-            url = self.base_url + "/" + endpoint
-        else:
-            url = (self.base_url or "") + (endpoint or "")
+        url = f"{(self.base_url or '').rstrip('/')}/{(endpoint or '').lstrip('/')}"
```

Comment:
Honestly I’m not sure this is better. To many nested levels. Maybe at least split this into a few lines?

```suggestion
        base_url = (self.base_url or "").rstrip("/")
        endpoint = (endpoint or "").lstrip("/")
        url = f"{base_url}/{endpoint}"
```